javascript/code-challenges/challenges-11.test.js

```diff
@@ -0,0 +1,228 @@
+'use strict';
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1 - Review
+
+Write a function named transformToLis that, given an object, returns an array of the key value pairs as html list items.
+
+For example:
+{
+  name: 'bob',
+  age: 32
+}
+
+Becomes:
+[
+<li>name: bob</li>,
+<li>age: 32</li>
+]
+------------------------------------------------------------------------------------------------ */
+
+function transformToLis(obj) {
+  let nameTargeted = obj.name;
+  let ageTargeted = obj.age;
+  return [`<li>name: ${nameTargeted}</li>`, `<li>age: ${ageTargeted}</li>`];
+}
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function named count that, given an integer and an array of arrays, uses either filter, map, or reduce to count the amount of times the integer is present in the array of arrays.
+
+Note: You might need to use the same method more than once.
+
+For example, count(5, [[1, 3, 5, 7, 9], [5, 5, 5], [1, 2, 3]]) returns 4.
+------------------------------------------------------------------------------------------------ */
+
+const count = (target, input) => {
+  let targetCount = 0;
+  input.forEach(subArray => {
+    targetCount = targetCount + (subArray.filter(indexValue => indexValue === target)).length;
+  });
+  return targetCount;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function that, given an array of integer arrays as input, calculates the total sum of all the elements in the array.
+
+You may want to use filter, map, or reduce for this problem, but are not required to. You may need to use the same method more than once.
+
+For example, [[1, 2, 3, 4, 5], [6, 7, 2, 4, 5, 7], [9, 2, 3, 6,]] returns 66.
+------------------------------------------------------------------------------------------------ */
+
+const totalSum = (input) => {
+  let total = 0;
+  input.forEach(array => {
+    total = total + array.reduce((sum, value) => sum + value, 0);
+  });
+  return total;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+Write a function named divisibleByFiveTwoToThePower that accepts an array of arrays as input.
+
+This function should first remove any elements that are not numbers or are not divisible by five.
+
+This function should then raise 2 to the power of the resulting numbers, returning an array of arrays.
+
+For example, [ [0,2,5,4], [2,4,10], [] ] should return [ [1, 32], [1024], [] ].
+------------------------------------------------------------------------------------------------ */
+
+const divisibleByFiveTwoToThePower = (input) => {
+  let divFilter = input.forEach(arr => {
+    arr.filter(element => element % 5 === 0);
+  });
```

Comment:
- For this one, the first part of the challenge tells us that we have to remove:
   - elements that are not divisible by 5
   - any elements that are not numbers...how can we get the data type of any given element using JavaScript? (this check is missing on line 77)
   👍